scripts/tools/silabs/FactoryDataProvider.py

```diff
@@ -0,0 +1,212 @@
+        passcode: Pairing passcode using in Spake2+
+        spake2_salt: Salt used to generate Spake2+ verifier
+
+    Returns:
+        The generated verifier string
+    """
+
+    cmd = [
+        args.gen_spake2p_path, 'gen-verifier',
+        '--iteration-count', str(args.spake2_iteration),
+        '--salt', args.spake2_salt,
+        '--pin-code', str(args.passcode),
+        '--out', '-',
+    ]
+    output = subprocess.check_output(cmd)
+    output = output.decode('utf-8').splitlines()
+    generation_results = dict(zip(output[0].split(','), output[1].split(',')))
+    return generation_results["Verifier"]
+
+
+def processArguments(args):
+    """ Do some checks on the received arguments.
+        Generate the Spake2+ verifier if needed and assign the values
+        to the global variables
+
+    Args:
+        The whole set of args passed to the script.
+    """
+    global Discriminator_ui
+    global Passcode_ui
+    global IterationCount_ui
+    global Salt_str
+    global Verifier_str
+    global mcuFamily
+    global jtagSerial
+
+    INVALID_PASSCODES = [00000000, 11111111, 22222222, 33333333, 44444444,
+                         55555555, 66666666, 77777777, 88888888, 99999999, 12345678, 87654321]
+
+    assert (bool(args.gen_spake2p_path) != bool(args.spake2_verifier)
+            ), "Provide either the spake2_verifier string or the path to the spake2 generator"
+    assert not (args.passcode in INVALID_PASSCODES), "The provided passcode is invalid"
+
+    print("passcode :", args.passcode)
+    print("discriminator :", args.discriminator)
+    print("spake2_iteration :", args.spake2_iteration)
+    print("spake2_salt :",  args.spake2_salt)
+    print("spake2_verifier :",  args.spake2_verifier)
+
+    Passcode_ui = args.passcode
+    Discriminator_ui = args.discriminator
+    IterationCount_ui = args.spake2_iteration
+    Salt_str = args.spake2_salt
+
+    if args.spake2_verifier:
+        Verifier_str = args.spake2_verifier
+    elif args.gen_spake2p_path:
+        Verifier_str = generate_spake2p_verifier(args)
+
+    if args.mcu_family:
+        mcuFamily = args.mcu_family
+
+    if args.jtag_serial:
+        jtagSerial = args.jtag_serial
+
+
+def add_SerialNo_To_CMD(cmdList):
+    """ Add the jtag serial command to the commander command
+
+        Args:
+        The commander command in list format
+    """
+    if jtagSerial:
+        cmdList.extend(["--serialno", jtagSerial])
+
+
+def create_nvm3injected_image():
+    """ Use commander command lines create a binary flashable to the EFR32 
+        containing the factory commissioning data in NVM3 section
+    """
+    global isDeviceConnected
+
+    # Retrieve the device current nvm3 data in a binary file
+    # It will be used as base to add the new credentials
+    inputImage = TEMP_FILE
+    cmd = ['commander', 'nvm3', 'read', '-o', inputImage, ]
+    add_SerialNo_To_CMD(cmd)
+    results = subprocess.run(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
```

Comment:
why this tool cannot be consolidated w the one created for credential + cd ?

Reply:
They do not target the same storage type and are not located in the same section in flash. Also, the logic is a bit different. It could be consolidated into one tool but it will then adds complexity to how to use both tools. I can look at doing that work after but I would want to get this in first